plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -251,44 +260,35 @@ func (p *plugin) Generate() (rm resmap.ResMap, err error) {
 		return nil, err
 	}
 
-	rm, err = p.h.ResmapFactory().NewResMapFromBytes(stdout)
-	if err == nil {
+	rm, resMapErr := p.h.ResmapFactory().NewResMapFromBytes(stdout)
+	if resMapErr == nil {
 		return rm, nil
 	}
 	// try to remove the contents before first "---" because
 	// helm may produce messages to stdout before it
-	stdoutStr := string(stdout)
-	if idx := strings.Index(stdoutStr, "\n---\n"); idx != -1 {
-		return p.h.ResmapFactory().NewResMapFromBytes([]byte(stdoutStr[idx:]))
+	r := &kio.ByteReader{Reader: bytes.NewBufferString(string(stdout)), OmitReaderAnnotations: true}
+	nodes, err := r.Read()
+
+	if len(nodes) != 0 {
+		rm, err = p.h.ResmapFactory().NewResMapFromRNodeSlice(nodes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse rnode slice into resource map: %w\n", err)
+		}
+		return rm, nil
 	}
-	return nil, err
+	return nil, fmt.Errorf("could not parse bytes into resource map: %w\n", resMapErr)
 }
 
 func (p *plugin) templateCommand() []string {
 	args := []string{"template"}
 	if p.ReleaseName != "" {
 		args = append(args, p.ReleaseName)
 	}
-	if p.Namespace != "" {
-		args = append(args, "--namespace", p.Namespace)
-	}
-	args = append(args, filepath.Join(p.absChartHome(), p.Name))
-	if p.ValuesFile != "" {
-		args = append(args, "--values", p.ValuesFile)
+	if p.Name != "" {
+		args = append(args, filepath.Join(p.absChartHome(), p.Name))
```

Comment:
Is the reason these two aren't extracted the private `absChartHome` for this one and the positional nature of ReleaseName?

Reply:
Yeah, I think the best alternative would be to pass the `absChartHome` value as an argument to `AsHelmArgs`, which in hindsight I think I actually prefer. So I can make that change if you agree.

Reply:
yeah, I agree with that change 👍

Reply:
done!